core/src/test/java/apoc/path/ExpandPathTest.java

```diff
@@ -525,4 +525,71 @@ public void testRelationshipFilterWorksWithoutTypeIncoming() {
 					assertEquals(1, maps.size());
 				});
 	}
+
+	@Test
+	public void testLabelWithSpecialChar() {
+		db.executeTransactionally(
+				"CREATE (n:`http://example.com/abc#Object` {one: 'alpha'})-[:REL]->(o:`http://www.w3.org/2002/07/owl#Class`:OwlClass {two: 'beta'})"
+		);
+
+		String pathFilter = ">|<";
+		// match using a single label (`http://www.w3.org/2002/07/owl#Class`)
+		String labelFilter = "/http\\://www.w3.org/2002/07/owl#Class|foo:bar|another";
+
+		Map<String, Object> config = Map.of("pathFilter", pathFilter,
+				"labelFilter", labelFilter);
+
+		TestUtil.testCall(db,
+				"MATCH (node:`http://example.com/abc#Object`) CALL apoc.path.expand(node, '>|<', '/http\\://www.w3.org/2002/07/owl#Class', 0, 2) " +
+						"YIELD path " +
+						"return distinct nodes(path) AS nodes",
+				config,
+				this::specialCharAssertions);
+
+		// with apoc.path.expandConfig
+		TestUtil.testCall(db,
+				"MATCH (node:`http://example.com/abc#Object`) CALL apoc.path.expandConfig(node, $config) " +
+						"YIELD path " +
+						"return distinct nodes(path) AS nodes",
+				Map.of("config", config),
+				this::specialCharAssertions);
+	}
+
+	@Test
+	public void testCompoundLabelWithSpecialChar() {
+		db.executeTransactionally(
+				"CREATE (n:`http://compound.example/abc#Object` {one: 'alpha'})-[:REL]->(o:`/http://www.w3.org/2002/07/owl#Class`:OwlClass:`Go:ku` {two: 'beta'})"
+		);
+
+		String pathFilter = ">|<";
+		// match using multiple labels (`http://www.w3.org/2002/07/owl#Class`, `OwlClass` and `Go:ku`)
+		String labelFilter = "//http\\://www.w3.org/2002/07/owl#Class:OwlClass:Go\\:ku|foo:bar|another";
```

Comment:
It would also be good with a test where you e.g. have a node with label \`foo:bar\` and make sure that it is not matched by a label filter foo:bar without the escaping \\\